src/coreclr/jit/emitxarch.cpp

```diff
@@ -8872,6 +8872,42 @@ void emitter::emitDispIns(
                 { // (val < 0)
                     printf("-0x%IX", -val);
                 }
+
+                unsigned iconType = id->idDebugOnlyInfo()->idFlags & GTF_ICON_HDL_MASK;
+                if ((val != 0) && id->idIsLargeCns())
+                {
+                    if (iconType == GTF_ICON_CLASS_HDL)
+                    {
+                        printf(" ; %.32s", emitComp->eeGetClassName(reinterpret_cast<CORINFO_CLASS_HANDLE>(val)));
+                    }
+                    else if (iconType == GTF_ICON_STR_HDL)
+                    {
+                        const WCHAR* str = emitComp->eeGetCPString(static_cast<size_t>(val));
+                        if (str != nullptr)
+                        {
+                            size_t    len            = wcslen(str);
+                            const int maxLength      = 32;
+                            WCHAR     buf[maxLength] = {0};
+                            wcsncpy(buf, str, min(maxLength, len));
```

Comment:
```suggestion
                            wcsncpy(buf, str, min(maxLength - 1, len));
```
I think you need `maxLength -1` here and below or else you'll overwrite the last null that you're depending on to null-terminate the string.